angular/projects/admin-nrpti/src/app/mines/mines-list/mines-list.component.ts

```diff
@@ -0,0 +1,210 @@
+
+/**
+ * Mine list page component.
+ *
+ * @export
+ * @class MinesListComponent
+ * @implements {OnInit}
+ * @implements {OnDestroy}
+ */
+@Component({
+  selector: 'app-mines-list',
+  templateUrl: './mines-list.component.html',
+  styleUrls: ['./mines-list.component.scss']
+})
+export class MinesListComponent implements OnInit, OnDestroy {
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public tableData: TableObject = new TableObject({ component: MinesTableRowComponent });
+  public tableColumns: IColumnObject[] = [
+    {
+      name: 'Name',
+      value: 'name',
+      width: 'col-3'
+    },
+    {
+      name: 'owner',
+      value: 'owner',
+      width: 'col-2'
+    },
+    {
+      name: 'Type',
+      value: 'type',
+      width: 'col-2'
+    },
+    {
+      name: 'Status',
+      value: 'status',
+      width: 'col-2'
+    },
+    {
+      name: 'Region',
+      value: 'region',
+      width: 'col-2'
+    },
+    {
+      name: '', // Edit button
+      value: '',
+      width: 'col-1',
+      nosort: true
+    }
+  ];
+
+  // Filters
+  public queryParams: Params;
+
+  constructor(
+    public location: Location,
+    public router: Router,
+    public route: ActivatedRoute,
+    public utils: Utils,
+    private loadingScreenService: LoadingScreenService,
+    private tableTemplateUtils: TableTemplateUtils,
+    private _changeDetectionRef: ChangeDetectorRef
+  ) {}
+
+  /**
+   * Component init.
+   *
+   * @memberof MinesListComponent
+   */
+  ngOnInit(): void {
+    this.loadingScreenService.setLoadingState(true, 'body');
+
+    this.route.params.pipe(takeUntil(this.ngUnsubscribe)).subscribe(params => {
+      this.queryParams = { ...params };
+      // Get params from route, shove into the tableTemplateUtils so that we get a new dataset to work with.
+      this.tableData = this.tableTemplateUtils.updateTableObjectWithUrlParams(params, this.tableData);
+
+      this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
+        if (!res || !res.mines) {
+          alert("Uh-oh, couldn't load NRPTI mines");
+          // mines not found --> navigate back to home
+          this.loadingScreenService.setLoadingState(false, 'body');
+          this.router.navigate(['/']);
+          return;
+        }
+
+        const mines = (res.mines[0] && res.mines[0].data && res.mines[0].data.searchResults) || [];
+        this.tableData.items = mines.map(mine => {
+          return { rowData: mine };
+        });
+
+        this.tableData.totalListItems =
+          (res.mines[0] &&
+            res.mines[0].data &&
+            res.mines[0].data.meta &&
+            res.mines[0].data.meta[0] &&
+            res.mines[0].data.meta[0].searchResultsTotal) ||
+          0;
+
+        this.tableData.columns = this.tableColumns;
+
+        this._changeDetectionRef.detectChanges();
+        this.loadingScreenService.setLoadingState(false, 'body');
+      });
+    });
+  }
```

Comment:
You have two subscriptions with data being set, but only one detectChanges.  Also redirecting to / is a bit harsh, showing 'no records found' along with the error in the page view instead of flicking around would probably be better UX.